src/libs/histogram.c

```diff
@@ -2146,79 +2135,100 @@ void gui_init(dt_lib_module_t *self)
+  dt_gui_add_class(d->button_box_main, "button_box");
+  gtk_widget_set_valign(d->button_box_main, GTK_ALIGN_START);
+  gtk_widget_set_halign(d->button_box_main, GTK_ALIGN_START);
+
+  // this intermediate box is needed to make the actions on buttons work
+  GtkWidget *box_left = gtk_box_new(GTK_ORIENTATION_HORIZONTAL, 0);
+  gtk_widget_set_valign(box_left, GTK_ALIGN_START);
+  gtk_widget_set_halign(box_left, GTK_ALIGN_START);
+  gtk_box_pack_start(GTK_BOX(d->button_box_main), box_left, FALSE, FALSE, 0);
+
+  d->button_box_opt = gtk_box_new(GTK_ORIENTATION_HORIZONTAL, 0);
+  dt_gui_add_class(d->button_box_opt, "button_box");
+  gtk_widget_set_valign(d->button_box_opt, GTK_ALIGN_START);
+  gtk_widget_set_halign(d->button_box_opt, GTK_ALIGN_END);
+
+  // this intermediate box is needed to make the actions on buttons work
+  GtkWidget *box_right = gtk_box_new(GTK_ORIENTATION_HORIZONTAL, 0);
+  gtk_widget_set_valign(box_right, GTK_ALIGN_START);
+  gtk_widget_set_halign(box_right, GTK_ALIGN_START);
+  gtk_box_pack_start(GTK_BOX(d->button_box_opt), box_right, FALSE, FALSE, 0);
+
+  d->button_box_rgb = gtk_box_new(GTK_ORIENTATION_HORIZONTAL, 0);
+  gtk_widget_set_valign(d->button_box_rgb, GTK_ALIGN_CENTER);
   gtk_widget_set_halign(d->button_box_rgb, GTK_ALIGN_END);
-
-  // FIXME: should histogram/waveform/vectorscope each be its own widget, and a GtkStack to switch between them?
-
-  // First two buttons choose scope type and view of that scope (if
-  // applicable). On click dt_lib_histogram_t data is updated,
-  // icons/tooltips are updated, and button sensitivity is set as
-  // needed.
+  gtk_box_pack_end(GTK_BOX(box_right), d->button_box_rgb, FALSE, FALSE, 0);
 
   // FIXME: the button transitions when they appear on mouseover (mouse enters scope widget) or change (mouse click) cause redraws of the entire scope -- is there a way to avoid this?
-  // FIXME: this could be a combobox to allow for more types and not to have to swap the icon on click
-  // icons will be filled in by _scope_type_update()
-  d->scope_type_button = dtgtk_button_new(dtgtk_cairo_paint_empty, CPF_NONE, NULL);
-  ac = dt_action_define(dark, NULL, N_("switch histogram mode"), d->scope_type_button, NULL);
-  dt_action_register(ac, NULL, _lib_histogram_change_mode_callback, 0, 0);
-  gtk_box_pack_start(GTK_BOX(d->button_box), d->scope_type_button, FALSE, FALSE, 0);
-  d->scope_view_button = dtgtk_button_new(dtgtk_cairo_paint_empty, CPF_NONE, NULL);
-  ac = dt_action_define(dark, NULL, N_("switch histogram type"), d->scope_view_button, NULL);
-  dt_action_register(ac, NULL, _lib_histogram_change_type_callback, 0, 0);
-  gtk_box_pack_start(GTK_BOX(d->button_box), d->scope_view_button, FALSE, FALSE, 0);
+
+  gchar *text;
+  for(int i=0; i<DT_LIB_HISTOGRAM_SCOPE_N; i++)
+  {
+    d->scope_type_button[i] = dtgtk_togglebutton_new(dt_lib_histogram_scope_type_icons[i], CPF_NONE, NULL);
+    text = g_strdup_printf("%s %s", _("set mode to"), _(dt_lib_histogram_scope_type_names[i]));
```

Comment:
@Mark-64 : After translation I find this **"set mode to"** superfluous. Just the name of the scope when flying over the icon seems enough. Ok to remove?

Reply:
ok